### PR TITLE
[NUI-1644][NUI-1669] Refactor error handling code for block transfer

### PR DESCRIPTION
**Pull request by tmathern**
## Description

Follow-up of https://github.com/adobe/node-httptransfer/pull/93.
- Refactor error code.
- Fix retry options sharing. 

Fixes #[NUI-1644](https://jira.corp.adobe.com/browse/NUI-1644) (partially), and subtask [NUI-1669](https://jira.corp.adobe.com/browse/NUI-1669).

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: lib/block/blocktransfer-error.js [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    }
+
+    /**
+     * Retrieves the error code representing the specific type of error. See ErrorCodes for more
+     * information.
+     *
+     * @returns {string} An error code value.
+     */
+    getCode() {
+        return this.code;
```
Comment (jdelbick) [inline lib/block/blocktransfer-error.js:61]: `.code` can still be used to access the code right? then why is this needed?

File: lib/block/blocktransfer-error.js [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+'use strict';
+
+const errorCodes = require('./error-codes');
```
Comment (jdelbick) [inline lib/block/blocktransfer-error.js:15]: IIUC, error-codes is used in both block and aem upload: https://github.com/adobe/node-httptransfer/blob/9ba21f111ad52e12080dd012f8a75fe29865db4a/lib/aem/aemupload.js#L34

so i think should `error-codes` file live somewhere not specific to block, wdyt?

**Review by jdelbick — APPROVED** [review]
LGTM

File: lib/block/blocktransfer-error.js (hunk 1 above)
Comment (tmathern) [reply lib/block/blocktransfer-error.js:61]: Yes, no change here, except where it's located.

File: lib/block/blocktransfer-error.js (hunk 2 above)
Comment (tmathern) [reply lib/block/blocktransfer-error.js:15]: Why not. A suggestion where?

**Review by jdelbick — APPROVED** [review]

File: lib/block/blocktransfer-error.js (hunk 2 above)
Comment (jdelbick) [reply lib/block/blocktransfer-error.js:15]: I guess at the root of lib https://github.com/adobe/node-httptransfer/tree/master/lib
